cirq/protocols/equal_up_to_global_phase_test.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import cirq
+
+
+def test_equal_up_to_global_phase_primitives():
+    assert cirq.equal_up_to_global_phase(1.0 + 1j, 1.0 + 1j, atol=1e-09)
+    assert cirq.equal_up_to_global_phase(1.0 + 1j, 1.0 - 1j, atol=1e-09)
+    assert cirq.equal_up_to_global_phase(np.exp(1j * 3.3),
+                                         1.0 + 0.0j,
+                                         atol=1e-09)
+    assert cirq.equal_up_to_global_phase(np.exp(1j * 3.3), 1.0, atol=1e-09)
+    assert cirq.equal_up_to_global_phase(np.exp(1j * 3.3), 1, atol=1e-09)
+    assert cirq.equal_up_to_global_phase(1j, 1, atol=1e-10)
+    # atol is determined by magnitude of complex vector, not components
+    assert cirq.equal_up_to_global_phase(0.0, 1e-10, atol=1e-09)
+    assert cirq.equal_up_to_global_phase(0, 0, atol=1e-09)
```

Comment:
You don't check any not-equal cases here. If the code said "if int then return True", this test would pass.

Cause the equality to fail both due to the tolerance getting higher, and due to the values being very different.